### ISSUE TITLE
v1.0.13 Crashing on DS214Play with DSM 7.1.1

### ISSUE THREAD

**Issue by smlomas** [eizedev/AirConnect-Synology#63]
I was running a v0.2xxx release ok, with the occasional restart needed. Initially just ran the upgrade to 1.0.13, from the DSM GUI,  but airupnp stops immediately after it reports it has started. Have uninstalled, then reinstalled, but still crashes out.

using : AirConnect-dsm7-x86-static-1.0.13-20221216.spk

[11:08:44] Starting AirConnect ...
[11:08:44] Linking airconnect configuration to "/volume1/airconnect/airconnect.conf".
[11:08:44] Linking airupnp configuration to "/volume1/airconnect/config.xml".
[11:08:44] Starting airupnp on 10.1.1.10:49154
[11:08:44] AirConnect successfully started on ip "10.1.1.10"!
[11:08:49] Status: airupnp not running!

**Comment by eizedev** [comment]
Which NAS model are you using?

**Comment by eizedev** [comment]
Ah, sorry, totally missed the topic. 214play is a 32bit arch (evansport), so https://github.com/eizedev/AirConnect-Synology/releases/download/1.0.13-20221216/AirConnect-dsm7-x86-1.0.13-20221216.spk should be the correct package.

I also have a 214play, i will also test it later this week.

**Comment by eizedev** [comment]
@tschniewind 
Do you have any log files or error messages from the upgrade to 1.0.13 that did not work?

Have you tried using the `AirConnect-dsm7-arm-1.0.13-20221216.spk` package without the static version?

I have no Synology device with ARM architecture, so i cannot reproduce the error or any error for this platform.

**Comment by tschniewind** [comment]
I have the same problem after upgrading a **RS814** from DSM 6.2 to DSM DSM 7.1.1-42962 Update 3 and then installing the AirConnect-dsm7-arm-static-**1.0.13**-20221216.spk.

After downgrading to AirConnect-dsm7-arm-static-**0.2.51.2**-20220206.spk everything works fine.

> 

There is no difference in using static or not. See the attached logfile written after a fresh install (removing the old package before). Even if the log says that AriConnect is started it is shown as stopped after a fiew seconds in the package manager. So there is unforunately no crash output.
[airconnect(13).log](https://github.com/eizedev/AirConnect-Synology/files/10375685/airconnect.13.log)

**Comment by tschniewind** [comment]
Yes indeed I found something in /var/log/messages:
```
2023-01-09T18:50:52+01:00 NAS2 coredump[12515]: Process airupnp[12514](/volume1/@appstore/AirConnect/airupnp) dumped core on signal [11]. Core file [/volume1/@airupnp.synology_armadaxp
_rs814.42962.core.gz]. Cmdline [/volume1/@appstore/AirConnect/airupnp -b 192.168.0.36:49154 -l 1000:2000 -x /volume1/@appstore/AirConnect/config.xml -o <NULL>,S1,S3,S5,S9,S12,ZP80,ZP90
,S15,ZP100,ZP120,1.0,LibreWireless,Fitzwilliam,2.2.6,AllShare1.0 -z -f /volume1/@appstore/AirConnect/log/airconnect.log -d all=info ]
2023-01-09T18:50:52+01:00 NAS2 coredump[12515]: Core file [/volume1/@airupnp.synology_armadaxp_rs814.42962.core.gz] size [45629]
2023-01-09T18:50:52+01:00 NAS2 coredump[12537]: Process aircast[12536](/volume1/@appstore/AirConnect/aircast) dumped core on signal [11]. Core file [/volume1/@aircast.synology_armadaxp
_rs814.42962.core.gz]. Cmdline [/volume1/@appstore/AirConnect/aircast -b 192.168.0.36 -l 1000:2000 -x /volume1/@appstore/AirConnect/config-cast.xml -z -f /volume1/@appstore/AirConnect/
log/airconnect.log -d all=info ]
2023-01-09T18:50:52+01:00 NAS2 coredump[12537]: Core file [/volume1/@aircast.synology_armadaxp_rs814.42962.core.gz] size [53658]
```

**Comment by tschniewind** [comment]
Yes indeed I found something in /var/log/messages:
```
2023-01-09T18:50:52+01:00 NAS2 coredump[12515]: Process airupnp[12514](/volume1/@appstore/AirConnect/airupnp) dumped core on signal [11]. Core file [/volume1/@airupnp.synology_armadaxp
_rs814.42962.core.gz]. Cmdline [/volume1/@appstore/AirConnect/airupnp -b 192.168.0.36:49154 -l 1000:2000 -x /volume1/@appstore/AirConnect/config.xml -o <NULL>,S1,S3,S5,S9,S12,ZP80,ZP90
,S15,ZP100,ZP120,1.0,LibreWireless,Fitzwilliam,2.2.6,AllShare1.0 -z -f /volume1/@appstore/AirConnect/log/airconnect.log -d all=info ]
2023-01-09T18:50:52+01:00 NAS2 coredump[12515]: Core file [/volume1/@airupnp.synology_armadaxp_rs814.42962.core.gz] size [45629]
2023-01-09T18:50:52+01:00 NAS2 coredump[12537]: Process aircast[12536](/volume1/@appstore/AirConnect/aircast) dumped core on signal [11]. Core file [/volume1/@aircast.synology_armadaxp
_rs814.42962.core.gz]. Cmdline [/volume1/@appstore/AirConnect/aircast -b 192.168.0.36 -l 1000:2000 -x /volume1/@appstore/AirConnect/config-cast.xml -z -f /volume1/@appstore/AirConnect/
log/airconnect.log -d all=info ]
2023-01-09T18:50:52+01:00 NAS2 coredump[12537]: Core file [/volume1/@aircast.synology_armadaxp_rs814.42962.core.gz] size [53658]
```

**Comment by tschniewind** [comment]
> @tschniewind Interesting! This is probably a bug of aircast/airupnp ARM binaries in connection to your architecture. I have created an issue for you with the information at [philippe44/AirConnect#446](https://github.com/philippe44/AirConnect/issues/446).

@eizedev Many thanks for your support. Hope that @philippe44 my help.

**Comment by tschniewind** [comment]
I think that armadaxp is base on armv7

**Comment by philippe44** [comment]
you're right - I need a log then (not a core dump)

**Comment by tschniewind** [comment]
where can I find the log you need?

**Comment by philippe44** [comment]
I'm not sure how @eizedev launches airconnect, but otherwise if you know how to launch it manually, it's -f <logfile>

**Comment by eizedev** [comment]
@philippe44 default command line (airupnp):

`/volume1/@appstore/AirConnect/airupnp -b 192.168.1.10:49154 -l 1000:2000 -x /volume1/@appstore/AirConnect/config.xml -o <NULL>,S1,S3,S5,S9,S12,ZP80,ZP90,S15,ZP100,ZP120,1.0,LibreWireless -z -f /volume1/@appstore/AirConnect/log/airconnect.log -d all info`

@tschniewind The log file is the `airconnect.log` file located in `/volume1/airconnect/log/airconnect.log`. It is a "mixed" log file with my logging events from the package and the log entries from airupnp/aircast to make it easier for the users.

You could also adjust the config in `/volume1/airconnect/airconnect.conf`. A log level change from info to debug:

Change `AIRUPNP_LOGLEVEL="all=info"` + `AIRCAST_LOGLEVEL="all=info"` to `AIRUPNP_LOGLEVEL="all=debug"` + `AIRCAST_LOGLEVEL="all=debug"` and restart the package. Therefore you do not need the SSH access.

Readme for the config part: https://github.com/eizedev/AirConnect-Synology#airconnectconf

**Comment by tschniewind** [comment]
Seems not to produce helpful infos. I deinstalled the 0.51-package an installed the 1.0.13-package. Then changed the loglevel for airupnp and started again, changed loglevel for aircast and started again. But after each try the package is not shown as started within the synology-ui and the each try after changing the loglevel logs first stopping ariConnect [17:15:15] & [17:19:39]:
[16:59:55] #### Start postinst... ####
[16:59:55] Creating airconnect config file at "/volume1/@appstore/AirConnect/airconnect.conf"
[16:59:55] Using installation type for both binaries (airupnp and aircast)
[16:59:55] Setting AIRUPNP_ENABLED=1 and AIRCAST_ENABLED=1 in "/volume1/@appstore/AirConnect/airconnect.conf"
[16:59:55] Setting AIRCAST_LATENCY="1000:2000" in "/volume1/@appstore/AirConnect/airconnect.conf"
[16:59:55] Setting AIRCAST_LOGLEVEL="all=info" in "/volume1/@appstore/AirConnect/airconnect.conf"
[16:59:55] Setting AIRUPNP_LATENCY="1000:2000" n "/volume1/@appstore/AirConnect/airconnect.conf"
[16:59:55] Setting AIRUPNP_LOGLEVEL="all=info" in "/volume1/@appstore/AirConnect/airconnect.conf"
[16:59:55] Setting AIRUPNP_PORT=49154 in "/volume1/@appstore/AirConnect/airconnect.conf"
[16:59:55] Setting FILTER_AIRPLAY2_DEVICES="<NULL>,S1,S3,S5,S9,S12,ZP80,ZP90,S15,ZP100,ZP120,1.0,LibreWireless,Fitzwilliam,2.2.6,AllShare1.0" in "/volume1/@appstore/AirConnect/airconne
ct.conf"
[16:59:55] Setting SYNO_IP="192.168.0.36" in "/volume1/@appstore/AirConnect/airconnect.conf"
[16:59:55] #### Done postinst ####
[16:59:58] Starting AirConnect ...
[16:59:58] Linking airconnect configuration to "/volume1/airconnect/airconnect.conf".
[16:59:58] Linking airupnp configuration to "/volume1/airconnect/config.xml".
[16:59:58] Starting airupnp on 192.168.0.36:49154
[16:59:58] Linking aircast configuration to "/volume1/airconnect/config-cast.xml".
[16:59:58] Starting aircast on 192.168.0.36
[16:59:59] AirConnect successfully started on ip "192.168.0.36"!
[17:15:15] Stopping AirConnect ...
[17:15:29] Starting AirConnect ...
[17:15:29] Linking airconnect configuration to "/volume1/airconnect/airconnect.conf".
[17:15:29] Linking airupnp configuration to "/volume1/airconnect/config.xml".
[17:15:29] Starting airupnp on 192.168.0.36:49154
[17:15:30] Linking aircast configuration to "/volume1/airconnect/config-cast.xml".
[17:15:30] Starting aircast on 192.168.0.36
[17:15:30] AirConnect successfully started on ip "192.168.0.36"!
[17:19:39] Stopping AirConnect ...
[17:19:49] Starting AirConnect ...
[17:19:49] Linking airconnect configuration to "/volume1/airconnect/airconnect.conf".
[17:19:49] Linking airupnp configuration to "/volume1/airconnect/config.xml".
[17:19:49] Starting airupnp on 192.168.0.36:49154
[17:19:50] Linking aircast configuration to "/volume1/airconnect/config-cast.xml".
[17:19:50] Starting aircast on 192.168.0.36
[17:19:50] AirConnect successfully started on ip "192.168.0.36"!

**Comment by philippe44** [comment]
That's not the log I'm looking for

**Comment by tschniewind** [comment]
This is taken from /volume1/@appstore/AirConnect/log/airconnect.log
Where to look else?

**Comment by eizedev** [comment]
that's only the log output from the installation/synology package, really strange.

Do you have SSH access to your NAS? Could you please stop the Airconnect-Synology package or kill the airupnp and aircast processes and run it manually?

With Logging Level Info:

`/volume1/@appstore/AirConnect/airupnp -b 192.168.0.36:49154 -l 1000:2000 -x /volume1/@appstore/AirConnect/config.xml -o <NULL>,S1,S3,S5,S9,S12,ZP80,ZP90,S15,ZP100,ZP120,1.0,LibreWireless -z -f /volume1/@appstore/AirConnect/log/airconnect.log -d all info`

That should produce any suitable log output in `/volume1/@appstore/AirConnect/log/airconnect.log`

**Comment by tschniewind** [comment]
FATAL: kernel too old
Illegal instruction (core dumped)

**Comment by eizedev** [comment]
Ah, that explains why no log messages for airupnp or aircast will be written @philippe44

**Comment by philippe44** [comment]
ah - In the update I've made to all my build tools, it includes min kernel and glibc. Having a kernel below 4.x is really a big risk

**Comment by tschniewind** [comment]
cat /proc/version
Linux version 3.2.101 (root@build2) (gcc version 8.5.0 (GCC) ) #42962 SMP Tue Oct 18 15:01:13 CST 2022

**Comment by philippe44** [comment]
oh, sorry about that but running on 3.2 is really really really bad

**Comment by tschniewind** [comment]
Doesn't come the linux kernel with the DSM-Installation?

**Comment by eizedev** [comment]
Synology does not update the kernel on older hardware models when updating to the newest DSM 7 Synology os release. 

This is really crappy from Synology. I have a similar situation on my NAS. I also still have kernel 3.10 despite the latest DSM version.

**Comment by eizedev** [comment]
Synology is applying security updates and other changes heavily in there DSM versions, but the kernel version still stays on old release

**Comment by tschniewind** [comment]
so there is no chance to get AirConnect run under this linux kernel?

**Comment by eizedev** [comment]
Could you please check your glibc version using `/lib64/libc.so.6 --version` or `/lib//libc.so.6 --version`?

@philippe44 Can you tell us the minimum versions of the kernel and glibc?

**Comment by philippe44** [comment]
> so there is no chance to get AirConnect run under this linux kernel?
>
You have to stick with older versions. More recent ones requires at least 4.x and changing that would mean rebuild all my toolchains

**Comment by eizedev** [comment]
@philippe44 I think the kernel is not the main problem, I think glibc is the showstopper.
I am running at kernel 3.10.xxx but the newest airupnp/aircast version is running:

```bash
/volume1/@appstore/AirConnect/airupnp -v
v1.0.13 (Dec 10 2022 @ 11:10:19)

/lib64/libc.so.6 --version
GNU C Library (GNU libc) stable release version 2.26, by Roland McGrath et al.
Copyright (C) 2017 Free Software Foundation, Inc.

uname -r
3.10.108
```

**Comment by philippe44** [comment]
> Could you please check your glibc version using `/lib64/libc.so.6 --version` or `/lib/libc.so.6 --version`?
> 
> @philippe44 Can you tell us the minimum versions of the kernel and glibc?
>
4.4.302 and 2.23

**Comment by tschniewind** [comment]
/lib/libc.so.6 --version
GNU C Library (GNU libc) stable release version 2.26, by Roland McGrath et al.
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.
Compiled by GNU CC version 8.5.0.
Available extensions:
	crypt add-on version 2.1 by Michael Glad and others
	GNU Libidn by Simon Josefsson
	Native POSIX Threads Library by Ulrich Drepper et al
	BIND-8.2.3-T5B
libc ABIs: UNIQUE
For bug reporting instructions, please see:
<http://www.gnu.org/software/libc/bugs.html>.

**Comment by eizedev** [comment]
interesting, I have an older kernel than 4.4 but "newer" glibc and on my system it is running.

**Comment by philippe44** [comment]
I'm not in front of an arm machine now but use ldd to check what I wrote

**Comment by eizedev** [comment]
ldd is not available on Synology devices, you could only the the version directly at the libraries running `/lib/libc.so.6 --version` or by running `lsof -p $$ | grep libc` (Result: /usr/lib/libc-2.26.so).

My system is also a x86_64 architecture. Perhaps this is the different to @tschniewind

**Comment by eizedev** [comment]
@tschniewind 
Unfortunately, I don't think you'll be able to get it to work, other than continuing to use the older version. 

You could stay at [0.2.51.2](https://github.com/eizedev/AirConnect-Synology/releases/tag/0.2.51.2-20220206)

I will update the documentation with this information.

@philippe44 is this "limitation" only for older ARMv7 architectures or also for other architectures? My x86_64 cpu arch with kernel 3.10. can still run airupnp and aircast.

**Comment by philippe44** [comment]
The limitation is on all Linux builds. I'm surprised that it runs on your machine, as all my compiler (I build them using crosstool-ng) have 4.4.320 and 2.23 set

**Comment by eizedev** [comment]
I just saw your config for crosstools-ng. Strange while it is running on my system. https://github.com/philippe44/cross-compiling/blob/8bf4f5b4e05ca5454540c7c0a4a80c8b4b7d7cfd/crosstools-ng-1.25.0/.config.stretch.x86_64#L282

I also have an older x86 Synology NAS. It has the same kernel version as @tschniewind but aircast 1.0.13 is still running on this old system:

```bash
root@Storebox:~# ./aircast-linux-x86
[19:54:13.559022] main:884 Starting aircast version: v1.0.13 (Dec 10 2022 @ 11:14:03)
[19:54:13.559313] main:891 no config file, using defaults
[19:54:13.559628] Start:662 Binding to 192.168.1.11
[19:54:13.930364] AddCastDevice:613 [0x8ec9320]: creating MAC
[19:54:13.930474] AddCastDevice:627 [0x8ec9320]: adding renderer (SHIELD) with mac CCCC85116FE6
^C[19:54:15.148203] DeleteCastDevice:379 [0x8ec9320]: Cast device stopped
^C^Croot@Storebox:~# ^C


root@Storebox:~# uname -a
Linux Storebox 3.2.101 #42661 SMP PREEMPT Mon Jun 27 15:08:01 CST 2022 i686 GNU/Linux synology_evansport_214play


root@Storebox:~# /lib/libc.so.6 --version
GNU C Library (GNU libc) stable release version 2.26, by Roland McGrath et al.
```

**Comment by philippe44** [comment]
It is strange indeed - I have to check exactly how kernel version is checked. I know how glibc is during loading, but I've never checked precisely re kernel. 

And don't worry, it will not change I don't want to rebuild all the tool chains and all the libraries that I need - it's a very cumbersome task.

**Comment by eizedev** [comment]
@smlomas als already mentioned, please install https://github.com/eizedev/AirConnect-Synology/releases/download/1.0.13-20221216/AirConnect-dsm7-x86-1.0.13-20221216.spk on your DS214play. I have just tested in on my DS214play, this version works without any problems.

**Comment by indywidualny** [comment]
For anyone having a problem with modern versions of the software. Simply use the older version. For example for Synology DS218j I found 0.2.51.2-20 version to be working alright, while all the recent ones are crashing with `[11:08:49] Status: airupnp not running!`

**Comment by eizedev** [comment]
synology fixed some bugs in glibc version, also on older hardware with older kernel. i additionally fixed a few long running bugs (and new ones) in latest version - it is again working on my old 415+ - perhaps give it a try:


## Release 1.11.3 is out — jumps from 1.8.3 to the current upstream

After a long quiet period, there's a new release: **[1.11.3-20260916](https://github.com/eizedev/AirConnect-Synology/releases/tag/1.11.3-20260916)**.

This brings two years of upstream AirConnect fixes/features (1.8.3 → 1.11.3, see the [upstream changelog](https://github.com/philippe44/AirConnect/blob/master/CHANGELOG)), plus several packaging bugs found and fixed along the way:

- **"Shows stopped in Package Center while it's actually running"** — `start-stop-status` used bare `ps` instead of `ps aux`, so daemonized processes were invisible to it. This should explain at least part of what's reported in #50 and #85 — not retroactively confirmed against those exact reports, but the mechanism matches.
- **Installation fails on Synology routers (SRM)** — every lifecycle script was tracked in Git as non-executable, which DSM tolerates but SRM does not. Fixed for both the DSM 7 and the legacy DSM 5/6 package lines.
- **IP field stays empty in the installer on SRM** — `grep -P` isn't supported by BusyBox and silently crashed there.
- **Wrong log file after upgrades** — `preupgrade`/`postupgrade` referenced a variable that was never actually set.
- **Fixed port range** (`AIRUPNP_PORTRANGE`) for anyone who needs that for firewall rules (#142).
- An automated upstream-update check, so the package doesn't fall two years behind again (#19).

Full changelog with all details and links: [CHANGELOG.md](https://github.com/eizedev/AirConnect-Synology/blob/master/CHANGELOG.md)

**Tested for real on a DS415+, a DS923+, and an RT2600ac (router/SRM)** — fresh install and an upgrade from 1.8.3. Not every architecture or model could be covered, so: **please test and report back here** whether it works for you or something's still off. One known open issue: the installer's automatic IP detection can pick the wrong interface on devices with multiple network interfaces (e.g. VPN/mesh) — the field is editable though, so it doesn't block installation.

Thanks for testing! 
René
